modules/sdk-core/src/bitgo/wallet/wallets.ts

```diff
@@ -554,9 +615,27 @@ export class Wallets implements IWallets {
     common.validateParams(params, ['walletShareId'], ['overrideEncryptedPrv', 'userPassword', 'newWalletPassphrase']);
 
     let encryptedPrv = params.overrideEncryptedPrv;
-
-    const walletShare = (await this.getShare({ walletShareId: params.walletShareId })) as any;
-
+    const walletShare = await this.getShare({ walletShareId: params.walletShareId });
+    if (walletShare.keychainOverrideRequired && walletShare.permissions.indexOf('admin') !== -1) {
+      if (_.isUndefined(params.userPassword)) {
+        throw new Error('userPassword param must be provided to decrypt shared key');
+      }
```

Comment:
Throwing here is misleading. Encrypted key data is only shared with people with `spend` permissions, not `admin`. You can technically be an admin and not a spender.

Reply:
> they will have encrypted key data, which holds true for two roles: admin and spenders.

This is not true for the amdin role. There is no gaurentee that the `admin` has the prv material. 

You need to check both admin and spender permissions here, not just admin.

Reply:
Done